manifests/init.pp

```diff
@@ -49,28 +55,45 @@
     default => $user
   }
 
+  $composer_group = $group ? {
+    'UNDEF' => $::composer::params::group,
+    default => $group
+  }
+
+  # common dependency
+  package { 'patch':
+    ensure => latest
+  }
+
+
+  # fetch composer into target_dir
   wget::fetch { 'composer-install':
     source      => $::composer::params::phar_location,
     destination => "${composer_target_dir}/${composer_command_name}",
-    execuser    => $composer_user,
+    execuser    => 'root',
```

Comment:
:-1: if you'd like to separate the downloading user from *$composer_user* then please create a new parameter